v2/pkg/image/image.go

```diff
@@ -59,7 +62,12 @@ func ParseRef(imgRef string) (ImageSpec, error) {
 	if strings.Contains(imgSpec.Name, "@") {
 		imgSplit := strings.Split(imgSpec.Name, "@")
 		if len(imgSplit) > 1 {
-			imgSpec.Digest = strings.Split(imgSplit[1], ":")[1]
+			validDigest, err := digest.Parse(imgSplit[1])
+			if err != nil {
+				return ImageSpec{}, fmt.Errorf("unable to parse image %s correctly, invalid digest: %v", imgRef, err)
+			}
+			imgSpec.Digest = validDigest.Hex()
```

Comment:
Hi @sherine-k,

It seems the Hex() func is deprecated. It is saying to use Digest.Encoded instead